src/Disks/ObjectStorages/IObjectStorage.h

```diff
@@ -73,13 +73,14 @@ class IObjectStorage
     ///     find . -type f
     ///
     /// @param children - out files (relative paths) with their sizes.
+    /// @param max_keys - useful to check existence
     ///
     /// NOTE: It makes sense only for real object storages (S3, Azure), since
     /// it is used only for one of the following:
     /// - send_metadata (to restore metadata)
     ///   - see DiskObjectStorage::restoreMetadataIfNeeded()
     /// - MetadataStorageFromPlainObjectStorage - only for s3_plain disk
-    virtual void findAllFiles(const std::string & path, RelativePathsWithSize & children) const;
+    virtual void findAllFiles(const std::string & path, RelativePathsWithSize & children, int max_keys) const;
```

Comment:
Why is it `int`? Isn't `size_t` better?

Reply:
`max_keys` for S3 and Azure is `int`:
- https://sdk.amazonaws.com/cpp/api/LATEST/class_aws_1_1_s3_1_1_model_1_1_list_objects_result.html#acfe96e989b8d180f3f4de259ac5c241e
- https://github.com/ClickHouse/azure-sdk-for-cpp/blob/ef75afc075fc71fbcd8fe28dcda3794ae265fd1c/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp#L185